src/server/index.js

```diff
@@ -110,13 +117,16 @@ function startServer () {
       const alignedIpAddress = ipAddress.padStart(15, ' ')
       const requestorMessage = `${flag} ${alignedIpAddress} requesting for ${targetAddress}`
       // check for greediness
+      //
+      // need to change network here:
       const balance = await ethQuery.getBalance(targetAddress, 'pending')
       const balanceTooFull = balance.gt(MAX_BALANCE)
       if (balanceTooFull) {
         console.log(`${requestorMessage} - already has too much ether`)
         return didError(res, new Error('User is greedy - already has too much ether'))
       }
       // send value
+      // need ot change network here as well:
```

Comment:
I don't think we're currently doing eip155 sigs here, we should fix that

Reply:
> also going to need infura api keys for each network

@kumavis I don't think we will if we're using Infura the way it normally is. As you can see here, you can change the Ethereum network from an Infura API simply by changing the subdomain:

<img width="652" alt="Screen Shot 2021-06-07 at 3 39 11 PM" src="https://user-images.githubusercontent.com/8144425/121085886-1115b300-c7a8-11eb-915b-775afed23a90.png">
<img width="627" alt="Screen Shot 2021-06-07 at 3 39 03 PM" src="https://user-images.githubusercontent.com/8144425/121085892-1410a380-c7a8-11eb-88af-306b98f5bfe5.png">

The one issue I'm having is how to dynamically change `rpcUrl` of the engine based on user's request. The relevant code is here:

https://github.com/ConsenSys-Academy/eth-faucet/blob/07dc818c341b2103175420295c068f25278a0e58/src/server/index.js#L40

Do you have any ideas of how to do this?